### DISCUSSION TITLE
Shared object between loader & action does not update after action is called?

### DISCUSSION THREAD

**Discussion by vedantroy** [remix-run/remix#2830] [Q&A] [answered]
I have a mock database for playing around with some loaders and actions.

Here's the rough layout:

```javascript
const db = { key: "bar" }

export const action = async ({ request }) => {
     db.key = "foo"
}

export const loader = async ({ request }) => {
    return json(db)
}
```

I have an issue though. When the action is called, it successfully updates `db.key`, however, the loader is called afterwards & the value is `{key: "bar" }`. Does anyone know why the object is not updated when the loader is called again?

**Comment by kiliman** [comment, marked as answer]
The issue is that in development, Remix is purging the require cache on every request. This is primarily to support LiveReload. However that also means all module-level variables are reset as well. 

The solution is to assign these module variables to the `global` object. 

You can see an example from the Jokes tutorial. 

https://remix.run/docs/en/v1/tutorials/jokes#connect-to-the-database
